src/backend/distributed/commands/function.c

```diff
@@ -565,10 +562,11 @@ EnsureFunctionCanBeColocatedWithTable(Oid functionOid, Oid distributionColumnTyp
  * UpdateFunctionDistributionInfo gets object address of a function and
  * updates its distribution_argument_index and colocationId in pg_dist_object.
  */
-static void
+void
 UpdateFunctionDistributionInfo(const ObjectAddress *distAddress,
 							   int *distribution_argument_index,
-							   int *colocationId)
+							   int *colocationId,
+							   bool localOnly)
```

Comment:
I think in my [original draft PR](https://github.com/citusdata/citus/pull/5132) the reason for this argument was to make sure we could call this function on the worker that pg_dist_object is propagated to: without causing propagation again. So, there's no concept op localOnly objects. All objects are tracked.

Now that I think about it again though, I think it would be better to use the value of `citus.enable_ddl_propagation` for this. Instead of having a `localOnly` argument. However, it could be that I tried that and didn't work for some reason I don't remember now.

Reply:
> So, there's no concept op localOnly objects. All objects are tracked.

Yes, thanks for the explanation. This is also how I see `localOnly` flag.

> Now that I think about it again though, I think it would be better to use the value of citus.enable_ddl_propagation for this.

I think we should **always** propagate the metadata changes to pg_dist_object. For example, we do not skip sending metadata changing commands on `citus_add_node` when   `enable_ddl_propagation = false`.

@JelteF what do you think on the above?

Reply:
So I'm not sure if I was completely clear. The reason to have this `localOnly` argument (afaik) is to make sure we don't get infinite recursion.

So in this case:
1. CREATE TYPE on node A
2. Call MarkObjectDistributed(..., false)
3. marks object distributed locally
4. Propagates to node B by calling citus_internal_add_object_metadata
5. On node B this calls MarkObjectDistributed(..., true)
6. marks object distributed locally

If 5 would use false, it would be propagated back to node A, and then back to node B and so on

My suggestion is to do this instead:

1. CREATE TYPE on node A
2. Call MarkObjectDistributed(...)
3. marks object distributed locally
4. Propagates to node B by setting enable_ddl_propagation=true and calling citus_internal_add_object_metadata,
5. On node B this calls MarkObjectDistributed(...)
6. marks object distributed locally, does not propagate because enable_ddl_propation is true

@onderkalaci 
> I think we should always propagate the metadata changes to pg_dist_object. 

If we don't propagate the CREATE TYPE or CREATE FUNCTION as well, then citus_internal_add_object_metadata would fail, because the type does not exist.